pl-engres-ipfs-stewards-roadmap.md

```diff
@@ -0,0 +1,200 @@
+
+Kubo has a bunch of battle-tested code over the years.  Unfortunately it’s not easy to consume as a library because of the repo sprawl and having to pull in all the relevant pieces.  It’s also hard for maintainers to keep it updated and we’re reliant on Kubo as the delivery vehicle to specify compatible versions.  Seeing the success go-libp2p has had moving to a monorepo, we will move to a similar pattern where Kubo specifics stay in Kubo but the generally usable parts for anyone writing an IPFS implementation in Go can live in libipfs
+
+Done: Core Kubo functionality is packaged into a library used both by Kubo as well as other implementers who want to support additional use cases without blocking on Kubo maintainers
+
+#### ⚫ Indexer double hashing support
+Theme: Privacy
+
+Define and deliver support for double hashing with indexers from Kubo to enhance IPFS privacy.
+
+Shared effort with Probelab, Bedrock, and libp2p.
+
+#### ⚫ Verifiable retrieval in light clients
+Theme: Ubiquitous Clients
+
+Deliver Rust library for client-side verifiable retrieval in light clients.
+
+Depends on:
+
+- Enable verifiable retrieval on the gateway
+
+Done:
+
+- Deliver Rust library for client-side verifiable retrieval in light clients
+
+Notes: 
+
+- We need to ensure we have alignment on use of Rust here with stakeholders
+- We believe it’s stategic for IP Stewards to invest in building muscle with Rust, and this is a good starting point
+
+### 2023-Q2
+#### ⚫ Implement default content routing selection
+Theme: Content Routing Innovation
+
+Depends on: 
+
+- Indexer Double Hashing Support
+- Spec for Content Routing Selection
+
+Done:
+
+- Full support (Kubo + any necessary infrastructure) to support content routing selection, including default use of cid.content by Kubo
+
+#### ⚫ “Slimmer Kubo”: Trim out unnecessary code
+Theme: Engineering Excellence
+
+Clean up old, non-critical functionality from Kubo to limit supported surface area and help team focus on strategic priorities/functionality.
+
+Kubo today has become a bit bloated with functionality that is not critical to our key use cases, has not been touched in a long time, etc. We then get issues opened against that functionality or questions about it, which diverts team time/resources away from key strategic work. We would like to clean some of this out of Kubo to help us be more efficient/effective, and focused on top priorities.
+
+Done: 
+
+1. Remove some specific things: Fuse, graphsync, others based on analysis
+    1. This actually requires a multi-part deprecation strategy, so we would start with deprecation, with code removal occurring potentially 6 months later.
+
+### Future
+#### ⚫ Hydras Plan
+Theme: Content Routing Innovcation
+
+Build long term strategy around hydras in general and hydra/indexer integration specifically for content routing performance.
+
+#### ⚫ Writeable Gateways
```

Comment:
I think its a good time to stop using "writable gateways" and focus on use case (which is onboarding data into IPFS) – thoughts?

```suggestion
#### ⚫ Data onboarding over HTTP POST
```